code/controllers/subsystem/nightshift.dm

```diff
@@ -31,11 +31,11 @@ SUBSYSTEM_DEF(nightshift)
 		nightshift = time < nightshift_end_time || time > nightshift_start_time
 
 	var/announcing = TRUE
-	if(high_security_mode && !emergency)
+	if(high_security_mode && !emergency && nightshift && !nightshift_active)
 		high_security_mode = FALSE
 		announce("Restoring night lighting configuration to normal operation.")

```

Comment:
So this message will run if the night shift ISN'T active??

Reply:
>Also probably not possible but: if(high_security_mode && !emergency && nightshift) should be if(high_security_mode && !emergency && nightshift && !nightshift_active) to prevent it from making the announcement but not actually doing an update. Again, should not be possible.

goddamn I just do what people tell me to do, yell at them not me

Reply:
If high security mode is on AND it is not an emergency AND it SHOULD be nightshift AND it is currently NOT, THEN make the announcement cause if by some miracle nightshift_active is already on later on in the code it won't do anything in the update but that announcement will still go off.

Honestly announcement should be a text argument that gets modified.

Reply:
This proc needs comments or the vars need to be better labeled, this is a headache for seemingly everyone.

Reply:
Yeah just realized, it does goof up by doing this... ugh.